#!/usr/bin/env python
PACKAGE = "x_rot_control"
vine_config = 0

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("pers_time_th", double_t, 0, "pers_time_th [s]",    10, 0,   20)
gen.add("pers_dist_th", double_t, 0, "pers_dist_th [m]",    0.2, 0,  2)
gen.add("pers_dist_th_same", double_t, 0, "pers_dist_th_same [m]",    0.1, 0,  0.5)
gen.add("consensus_th", int_t, 0, "consensus_th [-]",    10, 0,  100)
gen.add("speed_upper_lim", double_t, 0, "speed_upper_lim [m/s]",    0.5, 0,   1)
gen.add("num_of_sector", int_t, 0, "num_of_sector [-]",    180, 6,   360)
gen.add("window_size_param", int_t, 0, "window_size_param [-]",    10, 1,   360)
gen.add("obstacle_weight", double_t, 0, "obstacle_weight [-]",    0.95, 0,   1)
gen.add("target_dir_weight", double_t, 0, "target_dir_weight [-]",    0.2, 0,   1)
gen.add("prev_dir_weight", double_t, 0, "prev_dir_weight [-]",    0.3, 0,   1)
gen.add("inflation_radius", double_t, 0, "inflation_radius [m]",    0.7, 0.2,   1.5)
gen.add("direction_speed_lim", double_t, 0, "direction_speed_lim [deg/s]",    15, 1,   20)
gen.add("linear_accel_lim", double_t, 0, "linear_accel_lim [m/s^2]",    0.1, 0,   5)
gen.add("direction_gain_multi", double_t, 0, "direction_gain_multi [m]",    0.5, 0,   1)
gen.add("direction_gain_offset", double_t, 0, "direction_gain_offset [m]",    0.4, 0,   1)
gen.add("speed_gain_multi", double_t, 0, "speed_gain_multi [m]",    0.4, 0,   1)
gen.add("speed_gain_offset", double_t, 0, "speed_gain_offset [m]",    0.3, 0,   1)
gen.add("weight_inversion_lat_dist", double_t, 0, "weight_inversion_lat_dist [m]",    0.1, 0,   1)
gen.add("angle_to_goal_th", double_t, 0, "angle_to_goal_th [m]",    40, 0,   60)
gen.add("lateral_dist_th", double_t, 0, "lateral_dist_th [m]",    0.09, 0,   2)
gen.add("goal_dist_th", double_t, 0, "goal_dist_th [m]",    3, 0,   4)
gen.add("verbose", bool_t, 0, "verbose",    True)

if vine_config:
    gen.add("max_detection_dist", double_t, 0, "max_detection_dist: distance at which local planner takes control[m]",    3, 2,   10)
    gen.add("stop_distance", double_t, 0, "stop_distance [m]",    1.5, 1.2,   3)
    gen.add("max_angle_dist", double_t, 0, "max_angle_dist [deg]",    20, 0,   40)
    gen.add("bound_ang", double_t, 0, "bound_ang [deg]",    45, 0,   90)
else:
    gen.add("max_detection_dist", double_t, 0, "max_detection_dist: distance at which local planner takes control[m]",    4, 2,   10)
    gen.add("stop_distance", double_t, 0, "stop_distance [m]",    0.8, 1.0,   3)
    gen.add("max_angle_dist", double_t, 0, "max_angle_dist [deg]",    20, 0,   40)
    gen.add("bound_ang", double_t, 0, "bound_ang [deg]",    90, 0,   90)

exit(gen.generate(PACKAGE, "x_rot_control", "x_rot_control"))
